[HJ-PRK] iP

hj-prk opened the pull request.

Commit:
Add Gradle support

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-6

Commit:
A-Enums

Commit:
no message

Commit:
no message

Commit:
level 8

Commit:
no message

Commit:
no message

Commit:
Merge branch 'master' into branch-Level-8

Commit:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S2/ip

Commit:
Make the code more OOP

Commit:
Add a package

Commit:
Set up Gradle

Commit:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/Week2/src/main/Duke.java

Commit:
Create jar file

Commit:
Add javadoc

Commit:
Fix as the coding standard

Commit:
Fix as coding standard 2

Commit:
Merge branch 'master' into branch-A-JavaDoc

Commit:
Merge branch 'master' into branch-A-CodingStandard

Commit:
Apply find function to find a task by searching for a keyword

Commit:
Fix JavaDoc

Commit:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/Week2/src/main/Task.java

Commit:
Fix JavaDoc

Commit:
Merge branch 'master' into branch-Level-8

Commit:
Fix JavaDoc

Commit:
Fix as Coding Standard

Commit:
Fix as coding standard

Commit:
Merge branch 'master' into branch-Level-8

Commit:
Fix coding standard

Commit:
Merge branch 'master' into branch-A-CodingStandard

Commit:
Merge branch 'branch-Level-8'

Commit:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/Week2/src/main/Duke.java
#	src/main/java/Week2/src/main/Task.java

Commit:
Fix merge conflicts

Commit:
Fix merge conflicts

Commit:
Fix errors

hj-prk committed:
Update README.md

hj-prk committed:
Update README.md

hj-prk committed:
Update README.md

hj-prk committed:
Update README.md

hj-prk committed:
Update README.md

Commit:
Update gradle

Commit:
Update gradle

jamieeeleow commented on src/main/java/Duke.java:
```diff
@@ -1,4 +1,4 @@
-import java.util.Scanner;
+import java.util.*;
```
Perhaps the imported classes could be listed explicitly?

jamieeeleow commented on src/main/java/Duke.java:
```diff
@@ -1,5 +1,15 @@
 import java.util.*;
 public class Duke {
+    public static class task{
```
Maybe the class name could be ```Task``` instead of ```task``` with a small 't' to adhere to the coding standards where class names are written in PascalCase?

jamieeeleow commented on src/main/java/Duke.java:
```diff
@@ -55,8 +61,33 @@ public static void main(String[] args) {
-            } else {
-                tasklist.add(new task(tasklist.size()+1, comm, " "));
+            } else if(comm.startsWith("todo")) {
+                String doit = comm.substring(5, comm.length());
```
Maybe the naming of the variable can be changed to camelcase ```doIt``` instead 😄

jamieeeleow commented on src/main/java/Duke.java:
```diff
@@ -1,16 +1,22 @@
 import java.util.*;
 import java.lang.*;
+
+enum type {
```
I like the use of enum type!! 👍

jamieeeleow commented on src/main/java/Duke.java:
```diff
@@ -85,7 +95,7 @@ public static void main(String[] args) {
                 } else if (comm.startsWith("deadline")) {
                     String doit = comm.substring(9, comm.length());
                     String[] parts = doit.split("/by");
```
👍 use of the plural form for the list of String objects.

jamieeeleow commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+public class Event extends Task {
+
+    String from;
```
Perhaps modifiers could be added to the variables?

jamieeeleow commented on src/main/java/Duke.java:
```diff
@@ -1,5 +1,8 @@
 import java.util.*;
 import java.lang.*;
+import java.io.FileWriter;
+import java.io.IOException;
```
Maybe the ordering of import classes can be a tad bit more consistent 😃

jamieeeleow commented on src/main/java/Week2/src/main/Duke.java:
```diff
@@ -145,14 +145,19 @@ public static void main(String[] args) throws IOException {
 
+                } else if (comm.startsWith("find")) {
+                    String keyword = comm.substring(6, comm.length());
+                    Search sr = new Search(tasklist);
```
I like that the variable names are short, making the codes look neat and cleaner. However, the shortened names might be a bit confusing for different variables such as ```Scanner sc``` and ```Search sr``` since there is only one letter difference. Perhaps the names could be replaced with something more explicit? 😀

jamieeeleow commented on src/main/java/Week2/src/main/Deadline.java:
```diff
@@ -1,12 +1,27 @@
 package Week2.src.main;
+
+/**
+ * A subclass of Task
```
Maybe some punctuations should be added at the end of each description line?

jamieeeleow commented on src/main/java/Week2/src/main/Ui.java:
```diff
@@ -43,4 +34,18 @@ public void hello() {
+        Duke.lining();
+    }
+
+    public void showEmptyError() {
```
I like the naming of the methods that makes it really clear as to what the function does.

Commit:
Add GUI

Commit:
Add GUI

Commit:
Fix Error

Commit:
Fix Error

Commit:
Fix JavaDoc for GUI

Commit:
Fix JavaDoc for GUI

Commit:
Fix error

Commit:
Fix error

Commit:
Fix merge conflicts

Commit:
Change the general structure of the project to implement GUI.
Needs to change in order to invoke the Parser class from Duke.
The change allows to launch GUI.

Commit:
Add assertion

Commit:
Improve code quality

Commit:
Revert "Improve code quality"

This reverts commit 40a7f93de99e7494048ce0b24d43f4a0e0a61198.

Commit:
no message

Commit:
Improve code quality

hj-prk committed:
Merge pull request #1 from HJ-PRK/branch-A-Assertions

Branch a assertions

hj-prk committed:
Merge pull request #2 from HJ-PRK/branch-A-CodeQuality

Improve code quality

Commit:
no message

Commit:
Merge branch 'branch-A-Assertions'

Commit:
Add extension C-DetectDuplicates

Commit:
Improve code quality and OOP style
Add a new package Commands for a better viewing

Commit:
Add Ui screenshot

hj-prk committed:
Update README.md

hj-prk committed:
Update README.md

Commit:
Fix error on delete

Commit:
Merge branch 'master' of https://github.com/HJ-PRK/ip

Commit:
Fix bugs

Commit:
Fix bug and handle exceptions

Commit:
Fix error

Commit:
add test files
